### PR TITLE
:memo: Add TSDoc and Docusaurus support

### LINKED ISSUE #5003: Setup live documentation tool for the web3
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Request

We need to explore different tools available in the domain of code documentation, with following considerations. 

1. The tools and approach should be favourably towards live documentation. 
2. It should have some flexibility to mixup static docs 
3. It should be able to split the static docs file
4. The documentation should be publishable in different formats e.g. html, markup 

### Use Cases

- Document the main documentation of the web3 repo 
- Document individual packages within web3

### Implementation Ideas

- TypeDoc
- TsDoc

### Feature Examples/References

_No response_

### Are you willing to implement this feature?

- [X] Yes

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Comment by nikoulai** [comment]
When I run `yarn start` inside `docs` directory, I get 
```
[INFO] Starting the development server...
[ERROR] Loading of version failed for version current
[ERROR] Error: Invalid sidebar file at "sidebars.js".
These sidebar document ids do not exist:
- api/web3
...
```
If I remove items in sidebars.js  it runs successfully. Is there any error or it's on my side?
(I have run ./scripts/build_api_docs.js and yarn previously)

**Comment by nazarhussain** [comment]
@nikoulai Please make sure you are running with the latest branch. I just checked out locally and see following output. 

```
[INFO] Starting the development server...
Info: Loaded plugin typedoc-monorepo-link-types
Info: Loaded plugin typedoc-plugin-extras
Info: Loaded plugin @droppedcode/typedoc-plugin-relative-includes
Info: Loaded plugin typedoc-plugin-mdn-links
Info: Documentation generated at ./docs/api

[SUCCESS] Docusaurus website is running at http://localhost:3000/.
```

**Review by nikoulai — APPROVED** [review]

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+*/
+
+const lightCodeTheme = require('prism-react-renderer/themes/github');
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+	title: 'Web3.js',
+	tagline: 'The ultimate JavaScript library for Ethereum',
+	url: 'https://docs.web3js.com',
```
Comment (jdevcs) [inline docs/docusaurus.config.js]:
```suggestion
	url: 'https://docs.web3js.org',
```

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+					{
+						title: 'Community',
+						items: [
+							{
+								label: 'Stack Overflow',
+								href: 'https://stackoverflow.com/questions/tagged/web3.js',
+							},
+							{
+								label: 'Discord',
+								href: 'https://discord.gg/uH8JRuv5',
```
Comment (jdevcs) [inline docs/docusaurus.config.js]:
```suggestion
								href: 'https://discord.com/invite/pb3U4zE8ca',
```

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+							},
+						],
+					},
+				],
+				copyright: `Copyright © ${new Date().getFullYear()} Web3.js, Inc. Built with Docusaurus.`,
```
Comment (jdevcs) [inline docs/docusaurus.config.js]:
```suggestion
				copyright: `Copyright © ${new Date().getFullYear()} Web3.js . Built with Docusaurus.`,
```

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+								to: '/docs/guides/intro',
+							},
+						],
+					},
+					{
+						title: 'Community',
+						items: [
+							{
+								label: 'Stack Overflow',
+								href: 'https://stackoverflow.com/questions/tagged/web3.js',
```
Comment (jdevcs) [inline docs/docusaurus.config.js]: this results in: 
```
Questions tagged [web3.js]
0 questions
```

File: docs/docusaurus.config.js (hunk 2 above)
Comment (jdevcs) [reply docs/docusaurus.config.js]: this was resulting in invalid link

**Comment by jdevcs** [comment]
LGTM, except some minor changes required

File: docs/docusaurus.config.js (hunk 4 above)
Comment (nazarhussain) [reply docs/docusaurus.config.js]: Fixed the link.

**Review by jdevcs — APPROVED** [review]

**Pull request merged**